src/sagemaker/serve/builder/model_builder.py

```diff
@@ -577,6 +600,76 @@ def wrapper(*args, **kwargs):
 
         return wrapper
 
+    def _check_if_input_is_mlflow_model(self) -> bool:
+        """Checks whether an MLmodel file exists in the given directory.
+
+        Returns:
+            bool: True if the MLmodel file exists, False otherwise.
+        """
+        if self.inference_spec or self.model:
+            logger.info(
+                "Either inference spec or model is provided. "
+                "ModelBuilder is not handling MLflow model input"
+            )
+            return False
+
+        if not self.model_metadata:
+            logger.info(
+                "No ModelMetadata provided. ModelBuilder is not handling MLflow model input"
+            )
+            return False
+
+        path = self.model_metadata.get(MLFLOW_MODEL_PATH)
+        if not path:
+            logger.info(
+                "%s is not provided in ModelMetadata. ModelBuilder is not handling MLflow model "
+                "input",
+                MLFLOW_MODEL_PATH,
+            )
+            return False
+
+        # Check for S3 path
+        if path.startswith("s3://"):
+            s3_downloader = S3Downloader()
+            if not path.endswith("/"):
```

Comment:
Curious, are these validations for all the Aloy runs mentioned in the CX proposal?

we wanted to validate like this right?

```
def _validate(model_path):
    if os.path.exists(model_path):
        _process_local_path()
    elif model_path[:5] == "s3://":
        _process_s3_path()
    elif model_path[:6] == "runs:/":
        _process_mlflow_run_id()
    elif model_path[:8] == "models:/":
        _process_mlflow_model_registry_path()
    elif model_path[:4] == "arn:" and ":model-package/" in model_path:
        _process_model_package_arn()
    else:
        raise ValueError("invalid input")
```

Reply:
Right, in beta we only support for s3 input tho

Reply:
okk didn't know that...